### PR TITLE
Python scripts: plot customization

### PR DESCRIPTION
**Pull request by albestro**
This PR slightly changes how internally plot generation works, introducing also some new features.

- `_gen_nodes_plot` now returns `(fig, ax)`. Its role is just generating plots, it does not save on file anymore;
- introduced `NodePlotWriter`, a generator that using the previous function, allow to customize the plots via returned `(fig, ax)` and on exit it writes to specified file
- existing specialized functions (e.g. `gen_chol_plots`), now accepts additional parameters `customize_ppn` and `customize_time` which are functions accepting `(fig, ax)` that are internally used (if set, they are optional), to customize the respective plots

Apart from being able to customize plots using `(fig, ax)`, other new features are that `_gen_nodes_plot` accepts:
- `subplots_args`: a dict with keywords to pass to `plt.subplots()` call (e.g. `gridspec_kw`)
- `styles`: a list with a structure similar to `filts` (i.e. `(regex, value)`), but accepting as values a dict with matplotlib styles for `plot` call (e.g. color, linewidth, ...)
- `fill_area`: enable/disable the min-max area plot

note: I gave a try to `black` for formatting the `postprocess.py` file (separate commit). let me know if you like it or not.

### CODE REVIEW

File: scripts/postprocess.py [hunk 1]
```diff
@@ -12,8 +12,21 @@
 sns.set_theme()
 
 # plt_type : ppn | time
-def _gen_nodes_plot(plt_type, title, ylabel, file_name, df, logx, logy=False, combine_mb=False, filts=None, replaces=None):
-    fig, ax = plt.subplots()
+def _gen_nodes_plot(
```
Comment (rasolca) [inline scripts/postprocess.py:15]: As input parameters increased (and their value has to follow some constrains) I would improve the doc.

File: scripts/postprocess.py [hunk 2]
```diff
@@ -220,7 +276,16 @@ def calc_trsm_metrics(df):
     )
 
 
-def gen_chol_plots(df, logx=False, combine_mb=False, filts=None, replaces=None, filename_suffix=None):
+# customize_* functions should accept fig and ax as parameters
```
Comment (rasolca) [inline scripts/postprocess.py:279]: The same here... a little more doc doesn't hurt...

**Review by rasolca — COMMENTED** [review]
I would improve a bit the doc and/or add an example that uses the new interface.

**Comment by rasolca** [comment]
Regarding black: can you increase the line length to 105 chars as for C++ files?

**Comment by albestro** [comment]
> Regarding black: can you increase the line length to 105 chars as for C++ files?

In the last commit I applied the line-length=105 modifier, just on the file changed by this PR. In local I have also a simple config file for the entire project (if we want to pursue this path).

File: scripts/postprocess.py (hunk 1 above)
Comment (albestro) [reply scripts/postprocess.py:15]: Can you give it a quick check, please?

File: scripts/postprocess.py [hunk 3]
```diff
@@ -11,9 +11,33 @@
...
+    subplot_args=None,
+    fill_area=True,
+):
+    """
+    Args:
+        plt_type:       ppn | time
+        title:          title of the plot
+        df:             the pandas.DataFrame with the data for the plot
+        combine_mb:     group also by mb sizes
+        filts:          [regex] for matching benchmark names to filter from the plot
```
Comment (rasolca) [inline scripts/postprocess.py:32]: - is `[regex]` a standard way to document an array containing one ore more regex? I'm used to fact that square brackets indicate optional arguments.
- I would interpret `filter from` as `matching are excluded from`. Am I wrong?
   Maybe rewriting as following is more clear:
   `benckmarks matching one of these regular expressions will appear in the plot (None: all benchmarks will appear).`

File: scripts/postprocess.py [hunk 4]
```diff
@@ -11,9 +11,33 @@
...
+    styles=None,
+    subplot_args=None,
+    fill_area=True,
+):
+    """
+    Args:
+        plt_type:       ppn | time
+        title:          title of the plot
+        df:             the pandas.DataFrame with the data for the plot
+        combine_mb:     group also by mb sizes
```
Comment (rasolca) [inline scripts/postprocess.py:31]:
```suggestion
        combine_mb:     bool indicates if different mb has to be included in the same plot
```

File: scripts/postprocess.py (hunk 3 above)
Comment (albestro) [reply scripts/postprocess.py]: Good points. I tried to address your comments. Please, give it a look 😉

**Review by rasolca — APPROVED** [review]

**Pull request merged**